### ISSUE TITLE
Example of zval_get_tmp_string/zend_tmp_string_release seems incorrect

### ISSUE THREAD

**Issue by camporter** [phpinternalsbook/PHP-Internals-Book#100]
From https://www.phpinternalsbook.com/php7/zvals/casts_and_operations.html#casts

```php
zend_string *tmp_str;
zend_string *str = zval_get_tmp_string(val, &tmp_str);
// Do something with str.
zend_tmp_string_release(str);
```

It seems like this should instead be:
```php
zend_string *tmp_str;
zend_string *str = zval_get_tmp_string(val, &tmp_str);
// Do something with str.
zend_tmp_string_release(tmp_str);
```

which is also how the `tmp_string` is handled further down on line 128 and how it is used within PHP. Is that correct?

**Comment by nikic** [comment]
Yeah, that's a typo.

**Comment by nikic** [comment]
Fixes by #110.
